src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -1894,6 +1894,22 @@ export function pickClinicalDataColors(
     };
 }
 
+export function pickNewColorForClinicData(
+    d: ClinicalDataCount[],
+    usedColors: Set<string>,
+    colors: string[] = COLORS
+) {
+    let availableColors: string[] = _.cloneDeep(colors);
+    for (let item of usedColors) {
+        let index = availableColors.indexOf(item);
+        if (index > -1) availableColors.splice(index, 1);
+    }
+    const fixedColors = pickClinicalAttrFixedColors(d);
```

Comment:
lets keep an eye on this because I think that repeatedly cloning this array could be slow if this function is run many times. is there another way to do it, e.g. by rewriting `pickClinicalAttrColorsByIndex` to just skip over used colors?